rest-api-spec/src/main/resources/rest-api-spec/api/inference.inference.json

```diff
@@ -0,0 +1,45 @@
+{
+  "inference.inference": {
+    "documentation": {
+      "url": "https://www.elastic.co/guide/en/elasticsearch/reference/master/post-inference-api.html",
+      "description": "Perform inference"
+    },
+    "stability": "stable",
+    "visibility": "public",
+    "headers": {
+      "accept": ["application/json"],
+      "content_type": ["application/json"]
+    },
+    "url": {
+      "paths": [
+        {
+          "path": "/_inference/{inference_id}",
+          "methods": ["POST"],
+          "parts": {
+            "inference_id": {
+              "type": "string",
+              "description": "The inference Id"
+            }
+          }
+        },
+        {
+          "path": "/_inference/{task_type}/{inference_id}",
```

Comment:
Do you need this form as it is covered by `inference.text_embedding`, `inference.sparse_embedding` etc?

Reply:
I mainly wanted to include it for backward compatibility. It could prove helpful if a new task type is added to Elasticsearch but not added immediately in the specification, but a missing task type sounds much less likely than a missing provider. We could deprecate this form.

Reply:
> It could prove helpful if a new task type is added to Elasticsearch but not added immediately in the specification

True, let's leave it in for now if it becomes problematic it can be removed without breaking any existing usage as the `inference.{TASK_TYPE}` cases cover everything.